### PR TITLE
Adds DocumentDB Integration #2848

### LINKED ISSUE #2848: [CUSTOMER REQUEST] DocumentDB Integration
Be able to train timeseries models and get predictions from DocumentDB in both open source and MindsDB Cloud.

### PR DESCRIPTION
**Pull request by bispaul**
## Description

Please include a summary of the change and the issue it solves. 

Fixes #2848

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

## Testing DocumentDB Handler with [home_rentals.csv](https://github.com/mindsdb/mindsdb/files/12832142/home_rentals.csv)

Screenshot 2023-10-06 at 7 27 13 PM

**1. Starting a ssh tunnel to DocumentDB in AWS**
```
ssh -i "ec2docDBpf.pem" -L 27017:docdb-demo.cluster-c2cuphiyp2vn.ap-south-1.docdb.amazonaws.com:27017 ec2-user@ec2-3-110-104-64.ap-south-1.compute.amazonaws.com -N
```
**1. Testing CREATE DATABASE**

```
db.databases.insertOne({
    name: "example_documentdb",
    engine: "documentdb",
    connection_args: {
        "username": "username",
        "password": "password",
        "host": "127.0.0.1",
        "port": "27017",
        "database": "sample_database",
        "kwargs": {
            "directConnection": true,
            "serverSelectionTimeoutMS": 2000,
            "tls": true,
            "tlsAllowInvalidHostnames": true,
            "tlsAllowInvalidCertificates": true,
            "retryWrites": false,
            "tlsCAFile": "/home/global-bundle.pem"
        }
    }
});
```

create database

MONGO DBS

**2. Testing CREATE PREDICTOR**

```
db.predictors.insertOne({
     name: "home_rentals_docdb_model",
     predict: "rental_price",
     connection: "example_documentdb",
     select_data_query: "db.home_rentals.find({})"
});
```

CREATE_PREDICTOR

**3. Testing FIND STATUS OF PREDICTOR**

```
db.predictors.find({name: "home_rentals_docdb_model"});
```

PREDICTOR_STATUS

**4. Testing SELECT FROM PREDICTOR**

```
db.home_rentals_docdb_model.find({sqft: "823", location: "good", neighborhood: "downtown", days_on_market: "10"});
```

SCORING

**4. Testing DROP PREDICTOR**

```
db.predictors.deleteOne({name: "home_rentals_docdb_model"});
```

DROP

Drop a remark based on your observation.

-   [x] Works Great 💚 (This means that all the steps were executed successfuly and the expected outputs were returned.)
-   [ ] There's a Bug 🪲 [Issue Title](URL To the Issue you created) ( This means you encountered a Bug. Please open an issue with all the relevant details with the Bug Issue Template)

---

### CODE REVIEW

**Comment by bispaul** [comment]
Hi kindly let me know how to fix the MindsDB Tests & Deploy that is failing? @ZoranPandovski

**Comment by bispaul** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/integrations/handlers/documentdb_handler/Manual_QA.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Welcome to the MindsDB Manual QA Testing for DocumentDB Handler
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/documentdb_handler/Manual_QA.md:1]: Let's exclude Manual QA file, we are removing this files.

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Awesome, looks great. Please just excliude the Manual QA file

File: mindsdb/integrations/handlers/documentdb_handler/Manual_QA.md (hunk 1 above)
Comment (bispaul) [reply mindsdb/integrations/handlers/documentdb_handler/Manual_QA.md]: Removed the Manual QA

**Pull request merged**